ror-tools/src/main/scala/tech/beshu/ror/tools/RorToolsApp.scala

```diff
@@ -16,43 +16,82 @@
  */
 package tech.beshu.ror.tools
 
+import os.Path
 import scopt.OParser
 import tech.beshu.ror.tools.core.actions.*
 import tech.beshu.ror.tools.core.patches.base.EsPatch
 import tech.beshu.ror.tools.core.utils.{EsDirectory, RorToolsException}
 
-import scala.util.Try
+import scala.io.StdIn
+import scala.util.{Failure, Success, Try}
 
 object RorToolsApp {
 
   // todo:
   // 1. option: return success when already patched/unpatched
   // 2. restore backup when fails to patch
   def main(args: Array[String]): Unit = {
-    OParser
-      .parse(parser, args, Config(Command.Verify(None)))
-      .foreach { config =>
+    handle(args) match {
+      case Result.CommandNotParsed => ()
+      case Result.Success => ()
+      case Result.Failure(exitCode) => sys.exit(exitCode)
+    }
+  }
+
+  def handle(args: Array[String]): Result = {
```

Comment:
maybe we can move it to some trait? the handle method could be a protected one in this trait